### PR TITLE
add edit testing

### PR DESCRIPTION
**Pull request by manitheleo**

### CODE REVIEW

File: src/Components/Buttons/Button.test.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+test("when edit functionality clicks", async () => {
+  let item = "first todo item";
+  const handleEditClick = jest.fn();
+  render(<EditButton handleEditClick={handleEditClick} item={item} />);
+  userEvent.click(
+    screen.getByRole("button", {
+      name: /edit/i,
+    })
+  );
+  waitFor(() => {
```
Comment (ArashKianiamphora) [inline src/Components/Buttons/Button.test.js]: Why use `waitFor` ?

File: src/Components/Buttons/Button.test.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import { render, screen, waitFor } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import EditButton from "./EditButton";
+
+test("when edit functionality clicks", async () => {
```
Comment (ArashKianiamphora) [inline src/Components/Buttons/Button.test.js]: when writing test description please follow 

Should ..., Given ...

so for example in this case

Should return todo item current text, Given the user click on the edit button

File: src/Components/Buttons/Button.test.js (hunk 1 above)
Comment (manitheleo) [reply src/Components/Buttons/Button.test.js]: handleEditclick is call back function

Comment (ArashKianiamphora) [reply src/Components/Buttons/Button.test.js]: but do you need waitFor ?

**Pull request merged**